sdk/core/Azure.Core.TestFramework/src/ClientDiagnosticListener.cs

```diff
@@ -141,6 +146,21 @@ public void Dispose()
 
             foreach (ProducedDiagnosticScope producedDiagnosticScope in Scopes)
             {
+                var activity = producedDiagnosticScope.Activity;
+                var operationName = activity.OperationName;
+                // traverse the activities and check for duplicates among ancestors
+                while (activity != null)
+                {
+                    if (operationName == activity.Parent?.OperationName)
+                    {
+                        // Throw this exception lazily on Dispose, rather than when the scope is started, so that we don't trigger a bunch of other
+                        // erroneous exceptions relating to scopes not being completed/started that hide the actual issue
+                        throw new InvalidOperationException($"A scope has already started for event '{producedDiagnosticScope.Name}'");
+                    }
+
+                    _activityGraph.TryGetValue(activity, out activity);
```

Comment:
Why don't you do `activity = activity.Parent`?

Reply:
Not sure what I was thinking..